JadeMarble/Sources/TycheEndPointDetectorEngine.swift

```diff
@@ -97,44 +98,41 @@ public class TycheEndPointDetectorEngine {
                 return
             }
 
-            let engineState = ptrPcmData.withMemoryRebound(to: UInt8.self, capacity: Int(buffer.frameLength*2)) { (ptrData) -> Int32 in
+            let (engineState, inputData) = ptrPcmData.withMemoryRebound(to: UInt8.self, capacity: Int(buffer.frameLength * 2)) { (ptrData) -> (Int32, Data) in
                 #if DEBUG
-                self.inputData.append(ptrData, count: Int(buffer.frameLength)*2)
+                self.inputData.append(ptrData, count: Int(buffer.frameLength) * 2)
                 #endif
+                let inputData = Data(bytes: ptrData, count: Int(buffer.frameLength) * 2)
                 
                 // Calculate flushed audio frame length.
                 var adjustLength = 0
                 if self.flushedLength + Int(buffer.frameLength) <= self.flushLength {
                     self.flushedLength += Int(buffer.frameLength)
-                    return -1
+                    return (-1, inputData)
                 } else if self.flushedLength < self.flushLength {
                     self.flushedLength += Int(buffer.frameLength)
                     adjustLength = Int(buffer.frameLength) - (self.flushedLength - self.flushLength)
                 }
                 
-                return epdClientChannelRUN(
+                let engineState = epdClientChannelRUN(
                     self.engineHandle,
                     ptrData,
-                    myint(UInt32(buffer.frameLength) - UInt32(adjustLength))*2, // data length is double of frame length, because It is 16bit audio data.
+                    myint(UInt32(buffer.frameLength) - UInt32(adjustLength)) * 2, // data length is double of frame length, because It is 16bit audio data.
                     0
                 )
+                
+                return (engineState, inputData)
             }
-            guard 0 <= engineState else { return }
+            guard 0 <= engineState, let speexEncoder else { return }
```

Comment:
zero를 쓰면 조금더 있어보입니다.

```suggestion
            guard .zero <= engineState, let speexEncoder else { return }
```